### DISCUSSION TITLE
AGPL question regarding integrating SingleFile into an extension

### DISCUSSION THREAD

**Discussion by bryanph** [gildas-lormeau/SingleFile#1093] [Q&A] [answered]
I'm working on a service that (among other files) allows you to upload HTML files for annotation purposes, in particular self-contained HTML files like the ones you can generate with SingleFile. The current workflow is that the user would simply generate the HTML file with SingleFile and upload it manually into the app. However, it would of course be nicer if there exists custom extension with customized configuration so that I don't have to ask the users to configure their SingleFile extension a certain way and to upload the HTML file manually. 

You've provided some nice instructions on how to do so at https://github.com/gildas-lormeau/SingleFile/wiki/How-to-integrate-the-API-of-SingleFile-into-an-extension.

Here starts my question: would including the extension in this manner trigger the copy-left clause of AGPL and require me to use the same license in the created extension? I would assume so and that would be no problem. However, added onto that: if the extension could be configured to communicate with a server (this would then be the annotation server) wouldn't that then also copy-left all the way to the annotation server? AGPL seems pretty vague to me on this aspect (what constituties a derived work) and obviously I don't want to violate it's terms.

**Comment by bryanph** [comment]
I just realized this should have been in the discussions section, can it be moved?

**Comment by gildas-lormeau** [comment, marked as answer]
> would including the extension in this manner trigger the copy-left clause of AGPL and require me to use the same license in the created extension?

I confirm that from the moment you make a modification in the code of SingleFile, you must indeed use the same license and make the source code of the project where the code was modified available to your users.

> if the extension could be configured to communicate with a server (this would then be the annotation server) wouldn't that then also copy-left all the way to the annotation server? 

According to the FAQ, see https://www.gnu.org/licenses/gpl-faq.html#AGPLv3ServerAsUser, you're not forced to open source the code on server-side because the server cannot really be considered as a user interacting with your extension. I also confirm that I agree and that from my point of view you don't have to publish the server-side code (as long as it's not based on AGPL code or a similar license), and that's not my goal.
